api/v1alpha1/accessrequest_types.go

```diff
@@ -49,19 +50,19 @@ type AccessRequestSpec struct {
 	// TargetRoleName defines the role name the user will be assigned
 	// to once the access is approved
 	TargetRoleName string `json:"targetRoleName"`
-	// Application defines the Argo CD Application to assign the elevated
+	// AppProject defines the Argo CD AppProject to assign the elevated
 	// permission
-	Application TargetApplication `json:"application"`
+	AppProject TargetAppProject `json:"appProject"`
```

Comment:
In my opinion, it makes more sense to perform a request to have elevated permission for one application instead of a project. Then the controller is responsible to find the corresponding project of that application and manage the roles accordingly using the requested application fullname to generate the policies. Without applicationName, it is impossible to give permissions to a specific application and the only generated policies will be on all application on this project. We should design the Spec to allow us to target the least amount of privilege necessary.

Reply:
changed